frontend/src/components/SideNavBarButton.tsx

```diff
@@ -1,14 +1,33 @@
+"use client";
 import React from 'react'
+import {useRef} from 'react';
 
 interface SideNavBarButtonProps {
     svgIcon: React.ReactNode;
     text: string;
+    setCurrentSelection: (selection: string) => void;
+    selected: boolean;
 }
-export default function SideNavBarButton({svgIcon, text}: SideNavBarButtonProps) {
+export default function SideNavBarButton({svgIcon, text, setCurrentSelection, selected}: SideNavBarButtonProps) {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  
+  if(buttonRef.current)
```

Comment:
Nit: could simplify to `if (buttonRef.current) && selected) {}` and remove one level of nesting. but its ok as is too.

Reply:
copy that. Keep hitting me with the nits, gotta stay sharp 🔥

Reply:
To be honest Im not sure how else I'd avoid the  extra level of nesting in this case given the node reference could be null at some point, and doing  if(selected && buttonRef.current){}else{} would require an extra if inside the else statement for same reason. Is there something im overlooking? Im avoiding ternary operators, not saying it's your suggestion, because I know they're frowned upon for legibility reasons.

I'll go ahead and merge this, but do share your thoughts if it's not too much to ask.